# Project 1: Deaths by tuberculosis

by Duncan wekesa, 22 November 2018, edited 5 April 2016, updated 18 October and 20 December 2017 

This is the project notebook for the first part of The Open University's _Learn to code for Data Analysis_ course.

In 2000, the United Nations set eight Millenium Development Goals (MDGs) to reduce poverty and diseases, improve gender equality and environmental sustainability, etc. Each goal is quantified and time-bound, to be achieved by the end of 2015. Goal 6 is to have halted and started reversing the spread of HIV, malaria and tuberculosis (TB).
TB doesn't make headlines like Ebola, SARS (severe acute respiratory syndrome) and other epidemics, but is far deadlier. For more information, see the World Health Organisation (WHO) page <http://www.who.int/gho/tb/en/>.

Given the population and number of deaths due to TB in some countries during one year, the following questions will be answered: 

- What is the total, maximum, minimum and average number of deaths in that year?
- Which countries have the most and the least deaths?
- What is the death rate (deaths per 100,000 inhabitants) for each country?
- Which countries have the lowest and highest death rate?

The death rate allows for a better comparison of countries with widely different population sizes.

## The data

The data consists of total population and total number of deaths due to TB (excluding HIV) in 2013 in each of the BRICS (Brazil, Russia, India, China, South Africa) and Portuguese-speaking countries. 

The data was taken in July 2015 from <http://apps.who.int/gho/data/node.main.POP107?lang=en> (population) and <http://apps.who.int/gho/data/node.main.1317?lang=en> (deaths). The uncertainty bounds of the number of deaths were ignored.

The data was collected into an Excel file which should be in the same folder as this notebook.

In [2]:
import warnings
warnings.simplefilter('ignore', FutureWarning)

from pandas import *
data = read_excel('WHO POP TB all.xls')
data

Country  Population (1000s)  \
0                                          Afghanistan               30552   
1                                              Albania                3173   
2                                              Algeria               39208   
3                                              Andorra                  79   
4                                               Angola               21472   
5                                  Antigua and Barbuda                  90   
6                                            Argentina               41446   
7                                              Armenia                2977   
8                                            Australia               23343   
9                                              Austria                8495   
10                                          Azerbaijan                9413   
11                                             Bahamas                 377   
12                                             Bahrain                1332   
13                                          Bangladesh              156595   
14                                            Barbados                 285   
15                                             Belarus                9357   
16                                             Belgium               11104   
17                                              Belize                 332   
18                                               Benin               10323   
19                                              Bhutan                 754   
20                    Bolivia (Plurinational State of)               10671   
21                              Bosnia and Herzegovina                3829   
22                                            Botswana                2021   
23                                              Brazil              200362   
24                                   Brunei Darussalam                 418   
25                                            Bulgaria                7223   
26                                        Burkina Faso               16935   
27                                             Burundi               10163   
28                                      Côte d'Ivoire               20316   
29                                          Cabo Verde                 499   
..                                                 ...                 ...   
164                                           Suriname                 539   
165                                          Swaziland                1250   
166                                             Sweden                9571   
167                                        Switzerland                8078   
168                               Syrian Arab Republic               21898   
169                                         Tajikistan                8208   
170                                           Thailand               67010   
171          The former Yugoslav republic of Macedonia                2107   
172                                        Timor-Leste                1133   
173                                               Togo                6817   
174                                              Tonga                 105   
175                                Trinidad and Tobago                1341   
176                                            Tunisia               10997   
177                                             Turkey               74933   
178                                       Turkmenistan                5240   
179                                             Tuvalu                  10   
180                                             Uganda               37579   
181                                            Ukraine               45239   
182                               United Arab Emirates                9346   
183  United Kingdom of Great Britain and Northern I...               

## The range of the problem

The column of interest is the last one.

In [3]:
tbColumn = data['TB deaths']

The total number of deaths in 2013 is:

In [4]:
tbColumn.sum()

1072677.97

The largest and smallest number of deaths in a single country are:

In [5]:
tbColumn.max()

240000.0

In [6]:
tbColumn.min()

0.0

From less than 20 to almost a quarter of a million deaths is a huge range. The average number of deaths, over all countries in the data, can give a better idea of the seriousness of the problem in each country.
The average can be computed as the mean or the median. Given the wide range of deaths, the median is probably a more sensible average measure.

In [7]:
tbColumn.mean()

5529.267886597938

In [8]:
tbColumn.median()

315.0

The median is far lower than the mean. This indicates that some of the countries had a very high number of TB deaths in 2013, pushing the value of the mean up.

## The most affected

To see the most affected countries, the table is sorted in ascending order by the last column, which puts those countries in the last rows.

In [9]:
data.sort_values('TB deaths')

Country  Population (1000s)  TB deaths
147                             San Marino                  31       0.00
125                                   Niue                   1       0.01
111                                 Monaco                  38       0.03
3                                  Andorra                  79       0.26
129                                  Palau                  21       0.36
40                            Cook Islands                  21       0.41
118                                  Nauru                  10       0.67
76                                 Iceland                 330       0.93
68                                 Grenada                 106       1.10
5                      Antigua and Barbuda                  90       1.20
113                             Montenegro                 621       1.20
152                             Seychelles                  93       1.40
105                                  Malta                 429       1.50
143                  Saint Kitts and Nevis                  54       1.60
11                                 Bahamas                 377       1.80
14                                Barbados                 285       2.00
144                            Saint Lucia                 182       2.20
99                              Luxembourg                 530       2.20
44                                  Cyprus                1141       2.30
174                                  Tonga                 105       2.50
50                                Dominica                  72       2.70
137                                  Qatar                2169       2.70
179                                 Tuvalu                  10       2.80
145       Saint Vincent and the Grenadines                 109       3.10
126                                 Norway                5043       4.40
146                                  Samoa                 190       6.10
121                            New Zealand                4506       6.30
103                               Maldives                 345       7.60
12                                 Bahrain                1332       9.60
164                               Suriname                 539      12.00
..                                     ...                 ...        ...
160                            South Sudan               11296    4500.00
119                                  Nepal               27797    4600.00
2                                  Algeria               39208    5100.00
193                               Zimbabwe               14150    5700.00
184            United Republic of Tanzania               49253    6000.00
181                                Ukraine               45239    6600.00
46   Democratic People's Republic of Korea               24895    6700.00
4                                   Angola               21472    6900.00
158                                Somalia               10496    7700.00
31                                Cameroon               22254    7800.00
170                               Thailand               67010    8100.00
88                                   Kenya               44354    9100.00
163                                  Sudan               37964    9700.00
30                                Cambodia               15135   10000.00
100                             Madagascar               22925   12000.00
0                              Afghanistan               30552   13000.00
141                     Russian Federation              142834   17000.00
190                               Viet Nam               91680   17000.00
115                             Mozambique               25834   18000.00
159                           South Africa               52776   25000.00
116                                Myanmar               53259   26000.00
134                            Philippines               98394   27000.00
58                                Ethio

The table raises the possibility that a large number of deaths may be partly due to a large population. To compare the countries on an equal footing, the death rate per 100,000 inhabitants is computed.

In [14]:
populationColumn = data['Population (1000s)']
data['TB deaths (per 100,000)'] = tbColumn * 100000/ populationColumn
data

Country  Population (1000s)  \
0                                          Afghanistan               30552   
1                                              Albania                3173   
2                                              Algeria               39208   
3                                              Andorra                  79   
4                                               Angola               21472   
5                                  Antigua and Barbuda                  90   
6                                            Argentina               41446   
7                                              Armenia                2977   
8                                            Australia               23343   
9                                              Austria                8495   
10                                          Azerbaijan                9413   
11                                             Bahamas                 377   
12                                             Bahrain                1332   
13                                          Bangladesh              156595   
14                                            Barbados                 285   
15                                             Belarus                9357   
16                                             Belgium               11104   
17                                              Belize                 332   
18                                               Benin               10323   
19                                              Bhutan                 754   
20                    Bolivia (Plurinational State of)               10671   
21                              Bosnia and Herzegovina                3829   
22                                            Botswana                2021   
23                                              Brazil              200362   
24                                   Brunei Darussalam                 418   
25                                            Bulgaria                7223   
26                                        Burkina Faso               16935   
27                                             Burundi               10163   
28                                      Côte d'Ivoire               20316   
29                                          Cabo Verde                 499   
..                                                 ...                 ...   
164                                           Suriname                 539   
165                                          Swaziland                1250   
166                                             Sweden                9571   
167                                        Switzerland                8078   
168                               Syrian Arab Republic               21898   
169                                         Tajikistan                8208   
170                                           Thailand               67010   
171          The former Yugoslav republic of Macedonia                2107   
172                                        Timor-Leste                1133   
173                                               Togo                6817   
174                                              Tonga                 105   
175                                Trinidad and Tobago                1341   
176                                            Tunisia               10997   
177                                             Turkey               74933   
178                                       Turkmenistan                5240   
179                                             Tuvalu                  10   
180                                             Uganda               37579   
181                                            Ukraine               45239   
182                               United Arab Emirates                9346   
183  United Kingdom of Great Britain and Northern I...               

## Conclusions

The BRICS and Portuguese-speaking countries had a total of about 350 thousand deaths due to TB in 2013. The median shows that half of these coutries had fewer than 5,650 deaths. The much higher mean (over 29,000) indicates that some countries had a very high number. The least affected were Sao Tome and Principe and Equatorial Guinea, with 18 and 67 deaths respectively, and the most affected were China and India with 41 thousand and 240 thousand deaths in a single year. However, taking the population size into account, the least affected were Portugal and Brazil with less than 2.2 deaths per 100 thousand inhabitants, and the most affected were Guinea-Bissau and East Timor with over 70 deaths per 100,000 inhabitants.

One should not forget that most values are estimates, and that the chosen countries are a small sample of all the world's countries. Nevertheless, they convey the message that TB is still a major cause of fatalities, and that there is a huge disparity between countries, with several ones being highly affected.